### DISCUSSION TITLE
Swift heavy ion induced porosity and swelling

### DISCUSSION THREAD

**Discussion by cnotthoff** [idaholab/moose#17032] [Q&A General] [answered]
Hi moose community,
I was playing a bit with the phase field module to see if I can use moose to simulate the pore growth in GaSb under swift heavy ion irradiation [see here](https://arxiv.org/abs/1909.00563).
The phase field simulation seems to describe the pore nucleation and growth quite nice for an infinite sample. However, I got stuck trying to find a way to simulate a finite samples where the volume of the matrix is conserved and the sample shape changes (uni-axial swelling) in response to the increasing pore volume. As you can see in the paper, the pore shape changes quite a lot due to the macroscopic swelling.

Is there a way in moose to force the conservation of the matrix volume and model the swelling either by changing the mesh (using e.g. tensor_mechanics) or using a free surface on the top of a simulation volume?

Cheers,
Christian.

**Comment by dschwen** [comment, marked as answer]
Christian,
it sounds like tensor mechanics together with strain periodicity is what you need. 
https://mooseframework.inl.gov/source/actions/GlobalStrainAction.html
Global strain periodicity allows you to model representative volume elements with arbitrary deformation tensors applied to the overall simulation cell (uniaxial deformation, shear, volumetric expansion, etc.).
Daniel

**Reply by cnotthoff** [reply]
Hi Daniel,
thanks for your response. I will give it a try with global strain.
However, I'm not sure if I understand global strain correctly.

I would like to calculate the strain which result from the bubble evolution rather then applying a given strain. What I tried up to now was to create a volumetric strain term with a magnitude dependent on the matrix/bubble fraction so that the matrix volume should remains constant, but I didn't managed to get the volume really constant and after a few iterations the solver does not converge any more.

Cheers,
Christian.
